examples/Mediawiki/PageProtection.yaml

```diff
@@ -0,0 +1,170 @@
+    response:
+      body:
+        query:
+          tokens:
+            csrftoken: p/grab: token
+  # Create Protected Page
+  - inherit: edit
+    use-session: RootUser
+    request:
+      parameters:
+        title: Protected Page
+        createonly: true
+      form-data:
+        summary: create page
+        text: Protected page
+        token: p/var: token
+  # Create Semi Protected Page
+  - inherit: edit
+    use-session: RootUser
+    request:
+      parameters:
+        title: Semi Protected Page
+        createonly: true
+      form-data:
+        summary: create page
+        text: Semi Protected page
+        token: p/var: token
+  # Add edit protections to only allow sysops to edit Protected Page
+  - use-session: RootUser
+    request:
+      method: post
+      path: api.php
+      form-data:
+        action: protect
+        title: Protected Page
+        token: p/var: token
+        protections: edit=sysop
+        format: json
+    response:
+      body:
+        protect:
+          title: Protected Page
+          protections:
+            edit: sysop
+            expiry: infinite
+  # Add edit protections to only allow auto confirmed users to edit Semi Protected Page
+  - use-session: RootUser
+    request:
+      method: post
+      path: api.php
+      form-data:
+        action: protect
+        title: Semi Protected Page
+        token: p/var: token
+        protections: edit=autoconfirmed
+        format: json
+    response:
+      body:
+        protect:
+          title: Semi Protected Page
+          protections:
+            edit: autoconfirmed
+            expiry: infinite
+
+tests:
+  - description: Root user should be able to edit Protected Page
+    interaction:
+      - inherit: edit
+        use-session: RootUser
+        request:
+          parameters:
+            title: Protected Page
+          form-data:
+            summary: test edit
+            text: Protected Page edited by RootUser
+            token: p/var: token
+  - description: Root user should be able to edit Semi Protected Page
+    interaction:
+      - inherit: edit
+        use-session: RootUser
+        request:
+          parameters:
+            title: Semi Protected Page
+          form-data:
+            summary: test edit
+            text: Semi Protected Page edited by RootUser
+            token: p/var: token
+  - description: Autoconfirmed user should not be able to edit Protected Page
+    interaction:
+      - inherit: edit
+        use-session: Alice
+        request:
+            parameters:
+              title: Protected Page
+            form-data:
+              summary: test edit
+              text: Protected Page edited by Alice
+              token: p/var: Alice/edit-token
+          response:
+            body:
+              code: protectedpage
+              info: This page has been protected to prevent editing or other actions.
+  - description: Autoconfirmed user should be able to edit Semi Protected Page
```

Comment:
Alice's fixture will have to be modified to put her into the autoconfirmed group.
The user fixtures in my PR have examples for how to put a user into a group.

Reply:
Users are automatically added to the autoconfirmed group when their account satisfies the $wgAutoConfirmAge AND $wgAutoConfirmCount, which both default to 0. In this case, Alice is automatically included in the group. 

https://www.mediawiki.org/wiki/Manual:$wgAutoConfirmAge

Reply:
Oh, I thought the default was 100 or so,  and so we couldn't rely on it. Ok then.